librosa/feature/spectral.py

```diff
@@ -1313,10 +1347,11 @@ def chroma_stft(
     chromafb = filters.chroma(sr, n_fft, tuning=tuning, n_chroma=n_chroma, **kwargs)
 
     # Compute raw chroma
-    raw_chroma = np.dot(chromafb, S)
+    raw_chroma = np.einsum("cf,...ft->...ct", chromafb, S,optimize=True)
+    # raw_chroma = np.dot(chromafb, S)
 
     # Compute normalization factor for each frame
-    return util.normalize(raw_chroma, norm=norm, axis=0)
+    return util.normalize(raw_chroma, norm=norm, axis=-2)
```

Comment:
chroma_stft basically LGTM.  There is a subtlety here around `estimate_tuning` which I've already noted in this comment: https://github.com/librosa/librosa/pull/1351#issuecomment-894249191

We can leave this is for now, but I'll have to come back to it once we decide on how to handle tuning in multichannel chroma.